realm/realm-library/src/main/cpp/io_realm_internal_TableQuery.cpp

```diff
@@ -1739,9 +1740,10 @@ Java_io_realm_internal_TableQuery_nativeIsNotEmpty(JNIEnv *env, jobject, jlong n
         auto column_idx = static_cast<size_t>(index_arr[arr_len - 1]);
 
         // Support a backlink as the last column in a field descriptor
-        auto last = reinterpret_cast<Table *>(table_arr[arr_len-1]);
+        auto last = reinterpret_cast<Table*>(table_arr[arr_len-1]);
         if (last != nullptr) {
-            pQuery->and_query(src_table_ref->column<BackLink>(*last, column_idx).count() != 0);
+            TableRef last_ref = TableRef(last);
```

Comment:
I think this is a matter of taste.  I had an off-line discussion with @kneth about this and I agree with him that the explicit cast is better than my original code, which just used `last`, a `Table*`.  Being succinct is good, for sure.  I don't think this cast costs much of anything and, for me at least, it makes it clearer what's going on.   I'm good with either but, FWIW, I prefer this.